src/server/req_modify.c

```diff
@@ -848,90 +911,124 @@ req_modifyReservation(struct batch_request *preq)
 			case RESV_ATR_end:
 				temp = strtol(psatl->al_value, &end, 10);
 				if (!is_standing || temp < next_occr_start) {
-					send_to_scheduler = RESV_END_TIME_MODIFIED;
-					presv->ri_alter_etime = presv->ri_wattr[RESV_ATR_end].at_val.at_long;
-					presv->ri_alter_flags |= RESV_END_TIME_MODIFIED;
+					send_to_scheduler = 1;
+					presv->ri_alter.ra_etime = presv->ri_wattr[RESV_ATR_end].at_val.at_long;
+					presv->ri_alter.ra_flags |= RESV_END_TIME_MODIFIED;
 				} else {
-					resv_revert_alter_times(presv);
-					snprintf(log_buffer, sizeof(log_buffer), "%s", msg_stdg_resv_occr_conflict);
+					resv_revert_alter(presv);
 					log_event(PBSEVENT_RESV, PBS_EVENTCLASS_RESV, LOG_INFO,
-						preq->rq_ind.rq_modify.rq_objname, log_buffer);
+						  preq->rq_ind.rq_modify.rq_objname, msg_stdg_resv_occr_conflict);
 					req_reject(PBSE_STDG_RESV_OCCR_CONFLICT, 0, preq);
 					return;
 				}
 
 				break;
 			case RESV_ATR_duration:
-				send_to_scheduler = RESV_DURATION_MODIFIED;
-				presv->ri_alter_flags |= RESV_DURATION_MODIFIED;
+				send_to_scheduler = 1;
+				presv->ri_alter.ra_flags |= RESV_DURATION_MODIFIED;
+				break;
+			case RESV_ATR_resource:
+				if (strcmp(psatl->al_resc, "select") == 0) {
```

Comment:
if this strcmp fails, it means that someone submitted something like `pbs_rsub -lplace=scatter`, right?  Shouldn't this be an error?
Also, if you can flip the if statement, that'd be great.

Reply:
Good catch, pbs_ralter catches it, but if someone used the API, a -lplace=scatter would just have been ignored.